HTTP helper utility

dnlkoch opened the pull request:
Adds simple HTTP utility methods for `GET` and `POST`.

buehner commented on src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HTTPUtil.java:
```diff
@@ -0,0 +1,464 @@
+ *
+ */
+public class HTTPUtil {
+
```
I would prefer `HttpUtil` (instead of `HTTPUtil`) :wink:

buehner commented on src/shogun2-util/src/test/java/de/terrestris/shogun2/util/http/HTTPUtilTest.java:
```diff
@@ -0,0 +1,227 @@
+ */
+@SuppressWarnings("static-method")
+public class HTTPUtilTest {
+
```
`HttpUtilTest` instead of `HTTPUtilTest` !? :wink:

dnlkoch commented:
Thanks for your review @buehner!

I addressed your notes and moved the plain non-persisted model object `Response` to a new util package `model` additionally. I suggest to use this package for any upcoming internal models.

marcjansen commented on src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java:
```diff
@@ -0,0 +1,464 @@
+
+			if (httpStatus == HttpStatus.OK) {
+
+				response.setStatusCode(httpStatus);
```
I am not sure why we have this line.

marcjansen replied:
Ahh I see nowwe have our own wrapper class. Ignore the above note.

marcjansen commented on src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java:
```diff
@@ -0,0 +1,464 @@
+			try {
+				if (httpResponse != null) {
+					httpResponse.close();
+				}
```
Can we use closeQuietly?

marcjansen commented on src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java:
```diff
@@ -0,0 +1,464 @@
+	 */
+	private static HttpHost getSystemProxy(URI uri) throws UnknownHostException {
+
+		System.setProperty("java.net.useSystemProxies", "true");
```
I have always been wondering if we can set this only once, not for every call.

marcjansen commented:
Really great work, @dnlkoch! I have added som minor questions. Which you may want to look at.

Three more questions: 

* will https urls be fully supported? The code to determine a system wide proxy looks as if it is hardcoded for http.
* where will we handle allowed hosts? In the application?
* does this support other HTTP verbs besides POST and GET?

Again: a very well put together PR and the code looks really neat!
